src/analyzer/protocol/ftp/functions.bif

```diff
@@ -4,46 +4,47 @@ type ftp_port: record;
 %%{
 #include "zeek/Reporter.h"
 
-static zeek::RecordValPtr parse_port(const char* line)
+static zeek::RecordValPtr parse_port(std::string_view line)
 	{
 	auto r = zeek::make_intrusive<zeek::RecordVal>(zeek::BifType::Record::ftp_port);
 
-	int bytes[6];
-	if ( line && sscanf(line, "%d,%d,%d,%d,%d,%d",
+	bool good = false;
+	uint32_t port = 0;
+	uint32_t addr = 0;
+
+	int32_t bytes[6];
+	if ( line.size() >= 11 && sscanf(line.data(), "%d,%d,%d,%d,%d,%d",
```

Comment:
Apparently there's `SCNd32` as a format specifier - should we be explicit about the width given the change from `int` to `int32_t`?

https://stackoverflow.com/a/26618501

Reply:
Huh, never seen those before. I'll switch over to them.